projects/packages/my-jetpack/_inc/components/product-card/recommendation-actions.tsx

```diff
@@ -0,0 +1,51 @@
+import { Button, getProductCheckoutUrl } from '@automattic/jetpack-components';
+import clsx from 'clsx';
+import { useMemo } from 'react';
+import { getMyJetpackWindowInitialState } from '../../data/utils/get-my-jetpack-window-state';
+import useMyJetpackConnection from '../../hooks/use-my-jetpack-connection';
+import styles from './style.module.scss';
+import usePricingData from './use-pricing-data';
+
+const useUpsellLinks = ( slug: string, wpcomProductSlug: string ) => {
+	const { isUserConnected } = useMyJetpackConnection();
+
+	return useMemo( () => {
+		const { myJetpackUrl, siteSuffix } = getMyJetpackWindowInitialState();
+		const purchaseUrl = getProductCheckoutUrl(
+			wpcomProductSlug,
+			siteSuffix,
+			myJetpackUrl,
+			isUserConnected
```

Comment:
The purchase CTA seems to take me to the user-connect screen before checkout. All of our interstitials have the "connect-after-checkout" flag, we should probably have that here as well 🤔

Reply:
Good point, thanks!